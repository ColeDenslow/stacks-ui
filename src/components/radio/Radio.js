import React from 'react';
import cn from 'classnames';
import PropTypes from 'prop-types';

import './Radio.css';

function Radio(props) {
  const { className, checked, disabled, messageType, value, ...otherProps } = props;

  const classNames = cn([
    'stx-radio',
    { 'stx-radio--is-disabled': disabled },
    { 'stx-radio--is-checked': checked },
    { 'stx-radio--alert': messageType === 'alert' || messageType === 'error' },
    { 'stx-radio--warning': messageType === 'warning' },
    { 'stx-radio--success': messageType === 'success' },
    className,
  ]);

  return (
    <span
      className={classNames}
    >
      <input
        {...otherProps}
        role='radio'
        type='radio'
        value={value}
        className='stx-radio__input'
        checked={checked}
        disabled={disabled}
        aria-checked={checked}
        aria-disabled={disabled}
      />
      <span className='stx-radio__box' />
    </span>
  );
}

Radio.defaultProps = {};

Radio.propTypes = {
  /** Optional className for the chekcbox component */
  className: PropTypes.string,

  /** Whether or not the radio component is checked */
  checked: PropTypes.bool,

  /** Whether or not the radio component is disabled */
  disabled: PropTypes.bool,

  /** The type of message to display */
  messageType: PropTypes.oneOf(['success', 'warning', 'alert', '']),

  /** The value of the radio component */
  value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
};

export default Radio;
