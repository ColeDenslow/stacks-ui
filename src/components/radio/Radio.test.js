import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Radio from './Radio';

describe('Radio', () => {
  const onChange = () => 'onChange';
  const onClick = () => 'onClick';

  test('Radio renders with basic props', () => {
    const output = mount(<Radio
      className='the-class'
      onChange={onChange}
      onClick={onClick}
      tabIndex='1'
      name='name'
      value='value'
    />);

    expect(toJson(output)).toMatchSnapshot();
  });

  test('Radio renders with checked', () => {
    const output = mount(<Radio
      className='the-class'
      onChange={onChange}
      onClick={onClick}
      tabIndex='1'
      name='name'
      value='value'
      checked
    />);

    expect(toJson(output)).toMatchSnapshot();
  });

  test('Radio renders with disabled', () => {
    const output = mount(<Radio
      className='the-class'
      onChange={onChange}
      onClick={onClick}
      tabIndex='1'
      name='name'
      value='value'
      disabled
    />);

    expect(toJson(output)).toMatchSnapshot();
  });

  test('Radio renders with messageType = alert', () => {
    const output = mount(<Radio
      className='the-class'
      onChange={onChange}
      onClick={onClick}
      tabIndex='1'
      name='name'
      value='value'
      messageType='alert'
    />);

    expect(toJson(output)).toMatchSnapshot();
  });

  test('Radio renders with additional props', () => {
    const output = mount(<Radio
      className='the-class'
      onChange={onChange}
      onClick={onClick}
      tabIndex='1'
      name='name'
      value='value'
      data-additional-prop='additional-prop'
    />);

    expect(toJson(output)).toMatchSnapshot();
  });
});
