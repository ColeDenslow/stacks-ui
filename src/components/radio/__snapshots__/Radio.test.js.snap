// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio Radio renders with additional props 1`] = `
<Radio
  className="the-class"
  data-additional-prop="additional-prop"
  name="name"
  onChange={[Function]}
  onClick={[Function]}
  tabIndex="1"
  value="value"
>
  <span
    className="stx-radio the-class"
  >
    <input
      className="stx-radio__input"
      data-additional-prop="additional-prop"
      name="name"
      onChange={[Function]}
      onClick={[Function]}
      role="radio"
      tabIndex="1"
      type="radio"
      value="value"
    />
    <span
      className="stx-radio__box"
    />
  </span>
</Radio>
`;

exports[`Radio Radio renders with basic props 1`] = `
<Radio
  className="the-class"
  name="name"
  onChange={[Function]}
  onClick={[Function]}
  tabIndex="1"
  value="value"
>
  <span
    className="stx-radio the-class"
  >
    <input
      className="stx-radio__input"
      name="name"
      onChange={[Function]}
      onClick={[Function]}
      role="radio"
      tabIndex="1"
      type="radio"
      value="value"
    />
    <span
      className="stx-radio__box"
    />
  </span>
</Radio>
`;

exports[`Radio Radio renders with checked 1`] = `
<Radio
  checked={true}
  className="the-class"
  name="name"
  onChange={[Function]}
  onClick={[Function]}
  tabIndex="1"
  value="value"
>
  <span
    className="stx-radio stx-radio--is-checked the-class"
  >
    <input
      aria-checked={true}
      checked={true}
      className="stx-radio__input"
      name="name"
      onChange={[Function]}
      onClick={[Function]}
      role="radio"
      tabIndex="1"
      type="radio"
      value="value"
    />
    <span
      className="stx-radio__box"
    />
  </span>
</Radio>
`;

exports[`Radio Radio renders with disabled 1`] = `
<Radio
  className="the-class"
  disabled={true}
  name="name"
  onChange={[Function]}
  onClick={[Function]}
  tabIndex="1"
  value="value"
>
  <span
    className="stx-radio stx-radio--is-disabled the-class"
  >
    <input
      aria-disabled={true}
      className="stx-radio__input"
      disabled={true}
      name="name"
      onChange={[Function]}
      onClick={[Function]}
      role="radio"
      tabIndex="1"
      type="radio"
      value="value"
    />
    <span
      className="stx-radio__box"
    />
  </span>
</Radio>
`;

exports[`Radio Radio renders with messageType = alert 1`] = `
<Radio
  className="the-class"
  messageType="alert"
  name="name"
  onChange={[Function]}
  onClick={[Function]}
  tabIndex="1"
  value="value"
>
  <span
    className="stx-radio stx-radio--alert the-class"
  >
    <input
      className="stx-radio__input"
      name="name"
      onChange={[Function]}
      onClick={[Function]}
      role="radio"
      tabIndex="1"
      type="radio"
      value="value"
    />
    <span
      className="stx-radio__box"
    />
  </span>
</Radio>
`;
